sci-misc/boinc/boinc-7.20.2-r2.ebuild

```diff
@@ -23,18 +23,40 @@ fi
 
 LICENSE="LGPL-3"
 SLOT="0"
-IUSE="X cuda curl_ssl_gnutls +curl_ssl_openssl"
 
-REQUIRED_USE="^^ ( curl_ssl_gnutls curl_ssl_openssl ) "
+IUSE_VIDEO_CARDS="
+		video_cards_amdgpu
+		video_cards_intel
+		video_cards_nvidia
+		video_cards_radeonsi
+"
+
+IUSE="${IUSE_VIDEO_CARDS} X cuda curl_ssl_gnutls +curl_ssl_openssl cvmfs opencl"
+
+REQUIRED_USE="
+	^^ ( curl_ssl_gnutls curl_ssl_openssl )
+	cuda? ( video_cards_nvidia )
+	opencl? ( || ( ${IUSE_VIDEO_CARDS} ) )
+"
 
 # libcurl must not be using an ssl backend boinc does not support.
 # If the libcurl ssl backend changes, boinc should be recompiled.
 DEPEND="
 	acct-user/boinc
-	>=app-misc/ca-certificates-20080809
+	app-misc/ca-certificates
 	cuda? (
-		>=dev-util/nvidia-cuda-toolkit-2.1
-		>=x11-drivers/nvidia-drivers-180.22
+		x11-drivers/nvidia-drivers
+	)
+	cvmfs? (
```

Comment:
This strikes me as something that should rather use `optfeature.eclass`.

Reply:
Which one, cvmfs?

Reply:
Actually I am not a fan of optfeature.eclass. To be totally honest I absolutely despise it.
Here is why:

I want a particular feature to be used.

Classic way:
1. set USE flag
2. let portage handle the dependencies
3. enjoy
4. remove package
5. --depclean cleans up no longer needed dependencies.

optfeature.eclass approach:
1. emerge package
2. read message about optional runtime feature
3. emerge optional runtime feature providing package(s)
4. enjoy
5. remove package
6. optional runtime feature providing packages _NOT_ cleaned up, because they are recorded in world
7. said packages laying around maybe forever.

But that's just my personal opinion. Just today, when I switched from repoman to pkgdev, two of these "optional runtime feature" providing packages have been pointed out, and I have them now in my world file, although I do not need and/or use them myself. Just some random package that is a depeondency of another package that is a dependency of ... 
I hope you can see why I am no big fan...

Reply:
I believe the main issue with the current construct is that you will need to mask the cvmfs use flag on all arches that are not x86 or amd64. But I could be wrong, so we should get a second opinion.

I get your point on the drawbacks of the optfeature eclass, but if I am right, then it appears to be the lesser of the two evils.

Reply:
cvmfs only pulls in net-fs/cvmfs on x86 and amd64, pkgdev would have went haywire otherwise.
Can I *hide* the USE flag on other architectures?

Reply:
1. Yes, you can mask on particular arches.
2. optfeature is what it is. If you wish for an alternative, work on https://bugs.gentoo.org/453618 / https://bugs.gentoo.org/424283.

Reply:
Interesting, my answer got ... lost? It is no longer here.
Weird...

The cvmfs USE flag only pulls in net-fs/cvmfs on x86 and amd64. But it is visible on other arches, too, doing literally nothing.

So I agree, `optfeature.eclass` might be the better approach here.
I'll update the ebuilds shortly.